install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -103,3 +103,23 @@ spec:
             severity: warning
           annotations:
             message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90%. Master nodes starved of memory could result in degraded performance of the control plane."
+    - name: master-nodes-info-memory-usage
+      rules:
+        - alert: MasterNodesInfoMemoryUsage
+          expr: |
+            ((sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) - sum(node_memory_MemFree_bytes + node_memory_Buffers_bytes + node_memory_Cached_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" ))) / sum(node_memory_MemTotal_bytes AND on (instance) label_replace( kube_node_role{role="master"}, "instance", "$1", "node", "(.+)" )) * 100) > 40
+          for: 10m
+          labels:
+            severity: info
+          annotations:
+            message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 40%. Master nodes starved of memory could result in degraded performance of the etcd cluster."
```

Comment:
"master" -> "control plane", kubernetes/website#30284.  Also your `sum(...)` here are aggregating all control-plane nodes, so there's no `$labels.node`.  The current PromQL would be good for `HighOverallControlPlaneMemory`.  You want `sum by (instance) (...)` forms for `ExtremelyHighIndividualControlPlaneMemory` and similar.

Also, `message` has been replaced by requests for [`summary` and `description`][1], like:

```yaml
summary: >-
  Memory utilization across all control plane nodes is high, and could impact responsiveness and stability.
description: >-
  Given three control plane nodes, the overall memory utilization may only be about 2/3 of all available capacity.
  This is because if a single control plane node fails... blah blah... etcd and filesystem cache... blah..blah.
```

[1]: https://github.com/monitoring-mixins/docs/pull/10/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R183